src/hooks/useApproveErc20.ts

```diff
@@ -19,7 +19,9 @@ export const useApproveErc20 = (
   const callTransaction = async () => {
     const { data: signer } = await refetch()
     const erc20 = new Contract(tokenAddress, ERC20, signer)
-    return erc20.approve(spenderAddress, amountUnformatted)
+    const gasEstimate = await erc20.estimateGas.approve(spenderAddress, amountUnformatted)
+    const overrides: Overrides = !!gasEstimate ? { gasLimit: gasEstimate.mul(12).div(10) } : undefined
+    return erc20.approve(spenderAddress, amountUnformatted, overrides)
```

Comment:
Ethers actually uses the number of passed arguments when dealing with the overrides, so passing `undefined` will actually cause problems since it assumes that since something was passed, it's valid.
```suggestion
    if (!!overrides) return erc20.approve(spenderAddress, amountUnformatted, overrides)
    return erc20.approve(spenderAddress, amountUnformatted)
```